sequoia/methods/d3rlpy_methods/base_test.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+from d3rlpy.constants import ActionSpace
+
+from sequoia.methods.d3rlpy_methods.base import *
+from sequoia.settings.offline_rl.setting import OfflineRLSetting
+
+
+class BaseOfflineRLMethodTests:
+    Method: ClassVar[Type[BaseOfflineRLMethod]]
+
+    @pytest.fixture
+    def method(self):
+        return self.Method(train_steps=1, train_steps_per_epoch=1)
+
+    @pytest.mark.parametrize('dataset', OfflineRLSetting.available_datasets)
+    def test_offlinerl(self, method, dataset: str):
+
+        setting_offline = OfflineRLSetting(dataset=dataset)
+
+        #
+        # Check for mismatch
+        if isinstance(setting_offline.env.action_space, gym.spaces.Box):
+            if method.algo.get_action_type() is not ActionSpace.CONTINUOUS:
```

Comment:
Note: `ActionSpace.BOTH` would also work here.

Reply:
How could ActionSpace.BOTH work here? We're want to make sure that they directly match, ie. if env is continuous then algo is continuous (or both, but continuous is what matters) and the same for discrete.

Reply:
hmmm I don't think you got what I meant.
Algo with `ActionSace.BOTH` can be applied to both continuous and discrete.
Here you're saying that if the algo doesn't have `CONTINUOUS`, then it isn't applicable on a continuous env.
This isn't correct, see what I mean?
This should be:
```python3
if method.algo.get_action_type() not in {ActionSpace.CONTINUOUS, ActionSpace.BOTH}:
```
and vice-versa for discrete.

Reply:
Ahh I see

Reply:
Fixed